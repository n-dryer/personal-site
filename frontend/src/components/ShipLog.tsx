import React, { useRef } from 'react';
import { motion, Variants } from 'framer-motion';
import { useInView } from 'react-intersection-observer';
import { UserData } from '../types';

interface ShipLogProps {
  userData: UserData; // Keeping the UserData prop for future compatibility
}

export const ShipLog: React.FC<ShipLogProps> = ({ userData }) => {
  const containerRef = useRef<HTMLDivElement | null>(null);
  const [ref, inView] = useInView({
    threshold: 0.1,
    triggerOnce: false,
  });

  // Animation variants
  const containerVariants: Variants = {
    hidden: { opacity: 0 },
    visible: {
      opacity: 1,
      transition: {
        staggerChildren: 0.2,
        delayChildren: 0.3
      }
    }
  };

  const itemVariants: Variants = {
    hidden: { opacity: 0, y: 20 },
    visible: { 
      opacity: 1, 
      y: 0,
      transition: {
        type: "spring",
        stiffness: 50,
        damping: 15
      }
    }
  };

  return (
    <section 
      ref={ref} 
      id="shiplog" 
      className="py-16 bg-gray-50 dark:bg-gray-950 text-gray-800 dark:text-white mt-20 md:mt-32"
    >
      <div className="container mx-auto px-4">
        <h2 className="text-3xl font-bold mb-12 text-center">Professional Journey</h2>
        
        <motion.div 
          ref={containerRef}
          className="flex flex-col items-center justify-center py-16"
          variants={containerVariants}
          initial="hidden"
          animate={inView ? "visible" : "hidden"}
        >
          <motion.div
            className="bg-white/50 dark:bg-slate-900/50 backdrop-blur-sm
                      rounded-xl shadow-sm ring-1 ring-black/5 dark:ring-white/10
                      p-8 max-w-md text-center"
            variants={itemVariants}
          >
            <h3 className="text-xl font-bold mb-4 text-blue-600 dark:text-blue-400">
              Coming Soon
            </h3>
            <p className="text-gray-600 dark:text-gray-300">
              The professional journey timeline is currently being updated with a new interactive format.
              Check back soon to explore my career path in detail.
            </p>
            <div className="mt-6">
              <div className="inline-block w-10 h-1 bg-blue-500 rounded-full"></div>
            </div>
          </motion.div>
        </motion.div>
      </div>
    </section>
  );
};
