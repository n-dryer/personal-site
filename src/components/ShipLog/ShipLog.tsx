import React, { useRef } from 'react';
import { motion, Variants } from 'framer-motion';
import { useInView } from 'react-intersection-observer';
import { UserData } from '../../types'; // Adjusted import path

/**
 * Props for the ShipLog component.
 */
type ShipLogProps = {
  /** User data object. Currently unused but kept for future compatibility with a more detailed timeline. */
  userData: UserData;
}

/**
 * ShipLogComponent is intended to display a professional journey or a log of significant events/projects.
 * Currently, it renders a "Coming Soon" message as the interactive timeline feature is under development.
 * It uses framer-motion for animations triggered when the component scrolls into view.
 * This component is memoized for performance, as its re-render depends on the `userData` prop.
 *
 * @param {ShipLogProps} props - The properties passed to the component.
 * @returns {React.ReactElement} The rendered ShipLog section.
 * @see React.memo
 */
const ShipLogComponent = ({ userData: _userData }: ShipLogProps): React.ReactElement => {
  const containerRef = useRef<HTMLDivElement | null>(null);
  const [ref, inView] = useInView({
    threshold: 0.1,
    triggerOnce: false,
  });

  // Animation variants
  const containerVariants: Variants = {
    hidden: { opacity: 0 },
    visible: {
      opacity: 1,
      transition: {
        staggerChildren: 0.2,
        delayChildren: 0.3,
      },
    },
  };

  const itemVariants: Variants = {
    hidden: { opacity: 0, y: 20 },
    visible: {
      opacity: 1,
      y: 0,
      transition: {
        type: 'spring',
        stiffness: 50,
        damping: 15,
      },
    },
  };

  return (
    <section
      ref={ref}
      id='shiplog'
      className='mt-20 bg-bg-primary py-16 text-text-primary md:mt-32'
    >
      <div className='container mx-auto px-4'>
        <h2 className='mb-12 text-center text-3xl font-bold'>
          Professional Journey
        </h2>

        <motion.div
          ref={containerRef}
          className='flex flex-col items-center justify-center py-16'
          variants={containerVariants}
          initial='hidden'
          animate={inView ? 'visible' : 'hidden'}
        >
          <motion.div
            className='bg-surface max-w-md rounded-lg p-8 text-center shadow-sm'
            variants={itemVariants}
          >
            <h3 className='mb-4 text-xl font-bold text-accent'>
              Coming Soon
            </h3>
            <p className='text-text-secondary'>
              The professional journey timeline is currently being updated with
              a new interactive format. Check back soon to explore my career
              path in detail.
            </p>
            <div className='mt-6'>
              <div className='inline-block h-1 w-10 rounded-full bg-accent'></div>
            </div>
          </motion.div>
        </motion.div>
      </div>
    </section>
  );
};

/**
 * Memoized ShipLog component.
 * @see ShipLogComponent
 */
export const ShipLog = React.memo(ShipLogComponent); 