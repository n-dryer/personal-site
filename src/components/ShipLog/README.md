# ShipLog Component

This directory contains the `ShipLog` component, its tests, and related files.

## Purpose

The `ShipLog` component is intended to display a professional journey or a log of significant events/projects. Currently, it shows a "Coming Soon" message as the interactive timeline feature is under development.

## Usage

```tsx
import { ShipLog } from '@/components/ShipLog'; // Or the correct relative path
import { userData } from '@/data.mock'; // Or your actual data source

<ShipLog userData={userData} />;
```

## Props

- `userData`: `UserData` - An object containing user information. (Currently unused but kept for future compatibility).
