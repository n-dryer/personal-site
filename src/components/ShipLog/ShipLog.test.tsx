import React from 'react';
import { render, screen } from '@testing-library/react';
import { ShipLog } from './ShipLog'; // Updated import path

// Mock framer-motion as it's not essential for this basic test
jest.mock('framer-motion', () => ({
  motion: {
    div: ({ children, ...props }: React.PropsWithChildren<any>) => (
      <div {...props}>{children}</div>
    ),
  },
  useInView: jest.fn(() => true),
}));

// Mock any lucide-react icons that might be used
jest.mock('lucide-react', () => ({
  FileText: () => <div data-testid='file-text-icon'>FileText Icon</div>,
  X: () => <div data-testid='x-icon'>X Icon</div>,
  // Add other icons that might be used in ShipLog
}));

describe('ShipLog Component', () => {
  const mockUserData = {
    fullName: 'Test User',
    bioLine: 'Test Bio',
    photoUrl: '/test.jpg',
    email: 'test@example.com',
    phone: '(123) 456-7890',
    location: 'Test Location',
    socialLinks: [
      { name: 'GitHub', url: 'https://github.com/test' },
      { name: 'LinkedIn', url: 'https://linkedin.com/in/test' },
    ],
    resumeUrl: '/test_resume.pdf',
  };

  it('should render and display the coming soon message', () => {
    render(<ShipLog userData={mockUserData} />);
    expect(screen.getByText(/Coming Soon/i)).toBeInTheDocument();
    expect(screen.getByText(/professional journey timeline is currently being updated/i)).toBeInTheDocument();
  });
}); 