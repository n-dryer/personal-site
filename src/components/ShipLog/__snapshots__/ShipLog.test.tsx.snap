// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipLog Component renders correctly 1`] = `
<div>
  <section
    class="mt-20 bg-gray-50 py-16 text-gray-800 dark:bg-gray-950 dark:text-white md:mt-32"
    id="shiplog"
  >
    <div
      class="container mx-auto px-4"
    >
      <h2
        class="mb-12 text-center text-3xl font-bold"
      >
        Professional Journey
      </h2>
      <div
        animate="hidden"
        class="flex flex-col items-center justify-center py-16"
        data-testid="motion-div"
        initial="hidden"
        variants="[object Object]"
      >
        <div
          class="max-w-md rounded-xl bg-white/50 p-8 text-center shadow-sm ring-1 ring-black/5 backdrop-blur-sm dark:bg-slate-900/50 dark:ring-white/10"
          data-testid="motion-div"
          variants="[object Object]"
        >
          <h3
            class="mb-4 text-xl font-bold text-blue-600 dark:text-blue-400"
          >
            Coming Soon
          </h3>
          <p
            class="text-gray-600 dark:text-gray-300"
          >
            The professional journey timeline is currently being updated with a new interactive format. Check back soon to explore my career path in detail.
          </p>
          <div
            class="mt-6"
          >
            <div
              class="inline-block h-1 w-10 rounded-full bg-blue-500"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
